### PR TITLE
Add regression test for #5621

### PR DESCRIPTION
**Pull request by thehesiod**
## What do these changes do?
Adds regression test for issue #5621

## Related issue number
#5621

## Checklist

- [x] I think the code is well written
- [ ] Unit tests for the changes exist
- [ ] Documentation reflects the changes
- [ ] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep alphabetical order, the file is sorted by names.
- [ ] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` for example (588.bugfix)
  * if you don't have an `issue_id` change it to the pr id after creating the pr
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."

### CODE REVIEW

File: tests/test_web_urldispatcher.py [hunk 1]
```diff
@@ -515,3 +516,17 @@ async def test_static_absolute_url(aiohttp_client, tmpdir) -> None:
...
+async def test_decoded_match_regex(aiohttp_client) -> None:
+    app = web.Application()
+
+    async def handler(_):
+        return web.Response()
+
+    app.router.add_get("/{user_ids:(([0-9]+)|(u_[0-9a-f]{16}))(,(([0-9]+)|(u_[0-9a-f]{16})))*}/hello", handler)
+    client = await aiohttp_client(app)
+
+    r = await client.get(yarl.URL("/467%2C802%2C24834%2C24952%2C25362%2C40574/hello", encoded=True))
```
Comment (webknjaz) [inline tests/test_web_urldispatcher.py]: @thehesiod could you also add a test expecting that matching raw urlencoded URLs doesn't work?

File: tests/test_web_urldispatcher.py [hunk 2]
```diff
@@ -515,3 +516,17 @@ async def test_static_absolute_url(aiohttp_client, tmpdir) -> None:
...
     assert resp.status == 403
+
+
+async def test_decoded_match_regex(aiohttp_client) -> None:
+    app = web.Application()
+
+    async def handler(_):
+        return web.Response()
+
+    app.router.add_get("/{user_ids:(([0-9]+)|(u_[0-9a-f]{16}))(,(([0-9]+)|(u_[0-9a-f]{16})))*}/hello", handler)
```
Comment (webknjaz) [inline tests/test_web_urldispatcher.py]: Looks like this needs to be formatted for us to see the test run in the CI: https://github.com/aio-libs/aiohttp/pull/5633/checks?check_run_id=2384352929#step:6:120.

File: tests/test_web_urldispatcher.py [hunk 3]
```diff
@@ -515,3 +516,17 @@ async def test_static_absolute_url(aiohttp_client, tmpdir) -> None:
     client = await aiohttp_client(app)
     resp = await client.get("/static/" + str(fname))
     assert resp.status == 403
+
+
+async def test_decoded_match_regex(aiohttp_client) -> None:
```
Comment (webknjaz) [inline tests/test_web_urldispatcher.py:525]: Once that other test I requested below is added and it's recorded in the CI logs, please apply an xfail marker so we could incorporate this in master:
```suggestion
@pytest.mark.xfail(
    raises=AssertionError,
    reason="Regression in v3.7: https://github.com/aio-libs/aiohttp/issues/5621",
)
async def test_decoded_match_regex(aiohttp_client) -> None:
```

**Review by webknjaz — CHANGES_REQUESTED** [review]
I noticed that you're targeting the `3.7` branch but it doesn't get updates anymore. Please change the base to `master` as this is where the development goes. After it's merged into `master`, it can be backported into 3.8 as well.

File: tests/test_web_urldispatcher.py (hunk 3 above)
Comment (thehesiod) [reply tests/test_web_urldispatcher.py:525]: done

File: tests/test_web_urldispatcher.py (hunk 2 above)
Comment (thehesiod) [reply tests/test_web_urldispatcher.py]: done

**Comment by thehesiod** [comment]
I'm going to create a new PR against master

**Comment by thehesiod** [comment]
@webknjaz master version in https://github.com/aio-libs/aiohttp/pull/5635

**Pull request closed without merging**